libs/utils/src/CString.cpp

```diff
@@ -36,7 +36,6 @@ int StaticString::compare(const StaticString& rhs) const noexcept {
 UTILS_NOINLINE
 CString::CString(const char* cstr, size_t length) {
     if (length && cstr) {
-        assert(length == strlen(cstr));
```

Comment:
I removed this assert because this constraint doesn't seem necessary. CString works perfectly well for arbitrary blobs or alternative character encodings.

Reply:
okay, but that means that it's not really a  "c string" anymore. Could you maybe update the comments in the header that this is a supported use case?

Reply:
Good catch because a comment definitely needed to be updated.  Fixed.  Note that CString now has similar constraints to std::string, which can also hold arbitrary data when using the constructor that takes a length.